### ISSUE TITLE
Do the stable and nightly versions have any telemetry? 

### ISSUE THREAD

**Issue by annoying258** [sqlitebrowser/sqlitebrowser#1963]

**Comment by karim** [comment]
No.

We only check for new version on startup by connecting to https://raw.githubusercontent.com/sqlitebrowser/sqlitebrowser/master/currentrelease. It can be disabled from settings.

If I remember correctly, with the recent Microsoft's acquisition of GitHub, there were some concerns that Microsoft could use this information (how many users are using the application daily, on what operating system, where they are from, etc.) so we moved them to our own infrastructure.

This file is now hosted at https://download.sqlitebrowser.org/currentrelease and the stats are available at https://sqlitebrowser.org/stats/.

**Comment by annoying258** [comment]
Thanks for the meaningful reply. (⌒‿⌒)

---

> We only check for new version on startup

is that the first startup or every single startup? 

---

> It can be disabled from settings.

how to actually do this; could you please elaborate? to confirm; you mean the settings in the app, right?

I tried 
All options in Help
the 6 tabs in Edit->Preferences
All options in File
and all the other 2 options View and Tools

`result:  no settings or relevant option found.`


Maybe I'm tired. I'll try again later; when I have some time to learn about this app's location of settings for this mechanism.

---

Regarding the mechanism that controls it, if its in the source code; could you please post the link?

**Comment by annoying258** [comment]
Also; please don't rely solely on Patreon for funding. (ノωヽ)

I checked the sponsor button and there was only one link and it was to Patreon.
And no other links at all on the Readme. (°ロ°) !

**Comment by justinclift** [comment]
> is that the first startup or every single startup?

Every time.  It's purpose is to let people know when a new release is out. :smile:

It's only in Windows and macOS though, as they're the ones we create installers for.  On Linux/BSD/etc, the packages are installed using the OS utilities, so there's no need for the check.

This is the option name, if that helps:

![DB4S-Automatic_updates](https://user-images.githubusercontent.com/406299/62860070-661a4680-bd42-11e9-96ba-db21e6f10fbe.png)

> Regarding the mechanism that controls it, if its in the source code; could you please post the link?

Yep, it is.  Gimme a minute to find it... :smile:

**Comment by karim** [comment]
> Also; please don't rely solely on Patreon for funding. (ノωヽ)

I think this was discussed before in #1425. Patreon seems to be better than the alternative, but we have no problem adding more options if they were requested.

Do you know, or can recommend, an alternative to Patreon?

> I checked the sponsor button and there was only one link and it was to Patreon.
And no other links at all on the Readme. (°ロ°) !

Other than the Sponsor button on GitHub, there is a link to Patreon in the application menu (`Help > Donate on Patreon`). I'd say the button in GitHub provides more visibility than putting it inside the readme.

> Or maybe you can't use any other option. In which case, sorry to bother you.

We also have [Liberapay](https://liberapay.com/sqlitebrowser/) but we never talked about it (like Fight Club). :smile: 

I guess we forgot to mention it.

@justinclift It looks like we created two Liberapay accounts. Are both working? With all details set to receive funding? If yes then we should also add it in the Sponsor button too.

**Comment by justinclift** [comment]
Hmmm, I think @MKleusberg was doing stuff with them. :smile:

Haven't touched Liberapay personally. :wink:

**Comment by MKleusberg** [comment]
We seem to have two Liberpay accounts: https://liberapay.com/db4s (seems to be created by Justin) and https://liberapay.com/sqlitebrowser (created by me). The reason why we never talked about Liberapay is that we didn't set up any way to actually receive the money because it turned out they only support PayPal and Stripe and I didn't want to create an account with either of them :wink:

**Comment by justinclift** [comment]
Oh, I completely forgot that db4s account there.  Didn't remember setting it up... but now you mention it, I do remember something about not being happy with some app wanting permissions to every other repo in my account.  Which I then didn't proceed with.

**Comment by MKleusberg** [comment]
You're right. I remember now too :smile: I think they wanted access to all your organisations, which is why I created the other account because I only had the sqlitebrowser organisation anyway.

**Comment by MKleusberg** [comment]
@annoying258 If you want to turn off the automatic update check before starting DB4S for the first time, i.e. avoid even the first request to that ```currentrelease``` file, you can start DB4S with this command line option: ```--save-option checkversion/enabled=false```. It permanently disables the automatic update check before doing anything else.

**Comment by annoying258** [comment]
Well, I looked at the link from https://github.com/sqlitebrowser/sqlitebrowser/issues/1963#issuecomment-520377493 and it looks fine. Just need to check these first in the v3.11.2 tag.these must be the files->

https://github.com/sqlitebrowser/sqlitebrowser/blob/v3.11.2/src/MainWindow.cpp
https://github.com/sqlitebrowser/sqlitebrowser/blob/v3.11.2/src/RemoteDatabase.cpp

---

Patreon is on the top of my list of payment processors to avoid.

---

> something about not being happy with some app wanting permissions to every other repo in my account. Which I then didn't proceed with.

do they need that because Github's system for apps lacks granularity or what? And why don't all such apps have a page that lists the permission of their app along with the reason?

----

https://github.com/sqlitebrowser/sqlitebrowser/issues/1425#issuecomment-468273723

spoiler about flattr.com: its patreon without the newsworthy screw-ups, but the tos looks promising. unless you skip reading https://flattr.com/terms#Obligations

**Comment by karim** [comment]
Also, here is a link (https://help.github.com/en/articles/displaying-a-sponsor-button-in-your-repository#about-funding-files) of supported GitHub Sponsors. Maybe we can try one of these.

**Comment by justinclift** [comment]
> do they need that because Github's system for apps lacks granularity or what?

Not sure.  Regardless, there's no chance I'm (personally) going to add an app that I want specifically for 1 project/repo, when there's no way to limit it to just that.

With Patreon, I'm not personally all that bothered about them.  It's fairly common for startup organisations to go through teething periods where from the outside they look to be doing dumb stuff.  Hopefully they'll sort themselves out in a positive way over time, and it becomes a non-issue. :crossed_fingers:
